### PR TITLE
Rev

### PR DESCRIPTION
**Pull request by Adam-Dupaski**

### CODE REVIEW

File: content/howto/mobile/how-to-ota.md [hunk 1]
```diff
@@ -2,16 +2,16 @@
 title: "Release Over the Air Updates with App Center's CodePush"
 parent: "native-mobile"
 menu_order: 71
-description: A tutorial for pushing over the air updates (OTA) using App Center's CodePush.
```
Comment (Adam-Dupaski) [inline content/howto/mobile/how-to-ota.md:5]: "tutorial"

File: content/howto/mobile/how-to-ota.md [hunk 2]
```diff
@@ -2,16 +2,16 @@
...
+description: Describes pushing over the air updates (OTA) using App Center's CodePush.
 tags: ["native", "mobile", "ota", "native-builder", "over the air", "update"]
 ---
 
 ## 1 Introduction
 
-Using Native Mobile Builder and Mendix Studio Pro, you can update your Mendix Native Apps over the air (OTA).
```
Comment (Adam-Dupaski) [inline content/howto/mobile/how-to-ota.md:11]: "Native Apps" capitalized

File: content/howto/mobile/how-to-ota.md [hunk 3]
```diff
@@ -2,16 +2,16 @@
...
-OTA updates are a fast and painless way of updating things like layouts, pages, assets, or even you app's business logic (such as nanoflows and JavaScript actions).
```
Comment (Adam-Dupaski) [inline content/howto/mobile/how-to-ota.md:12]: "Things" too generic?
Typo

File: content/howto/mobile/how-to-ota.md [hunk 4]
```diff
@@ -2,16 +2,16 @@
...
+Using Native Mobile Builder and Mendix Studio Pro, you can update your Mendix native apps over the air (OTA).
+OTA updates are a fast and painless way of updating elements like layouts, pages, assets, or even your app's business logic (such as nanoflows and JavaScript actions).
 
-Native apps are separated into two parts: a wrapper that is basically a native iOS or Android app and a bundle that is being loaded dynamically by said wrapper. Things like your business logic and static assets are part of this dynamically-loaded bundle. When you have changes you want to deploy, the Native Mobile Builder can bundle them in a new, updated bundle and painlessly deploy them. On the next app restart, your app's users will be updated to the latest version and continue their business as usual.
```
Comment (Adam-Dupaski) [inline content/howto/mobile/how-to-ota.md:14]: Eliminate unnecessary adverbs
Make sentence more minimal/concise
"Your app's users"

File: content/howto/mobile/how-to-ota.md [hunk 5]
```diff
@@ -23,15 +23,16 @@ Currently OTA  does not update your app while the app is open or minimized.
...
+
 ## 2 Prerequisites {#prerequisites}
 
 Before starting this how-to, make sure you have completed the following prerequisites:
 
-* Install Mendix Studio Pro v8.15 or above using the online installer (the offline installer does not include the Mendix Native Mobile Builder dependency)
 * Deploy your app by completing [How to Deploy Your First Mendix Native Mobile App](deploying-native-app).
 * Ensure you have set up and configured App Center for your app. For information on setting up App Center, see the [App Center Token](deploying-native-app#appcenter-token) section of *How to Deploy Your First Mendix Native Mobile App*. If you previously opted out of configuring App Center while building your app, you must navigate to the **Tokens** screen and toggle on the **I want to use App Center** option.
 * Install your app on a test device or emulator.
-* Read the [Offline First Reference Guide](/refguide/offline-first).
```
Comment (Adam-Dupaski) [inline content/howto/mobile/how-to-ota.md:34]: Use only title in cross-reference text if cross-referencing a doc (as opposed to cross-referencing a concept, which is more flexible)

File: content/howto/mobile/how-to-ota.md [hunk 6]
```diff
@@ -23,43 +23,48 @@ Currently OTA  does not update your app while the app is open or minimized.
...
+* Read the [Offline-First](/refguide/offline-first) reference guide.
 
 ## 3 When to Use OTA Updates
 
-### 3.1 Safely Pushing OTA Updates Without Redeploying Your Mendix App {#safeToUpdate}
```
Comment (Adam-Dupaski) [inline content/howto/mobile/how-to-ota.md:38]: Section heading a little long

File: content/howto/mobile/how-to-ota.md [hunk 7]
```diff
@@ -23,43 +23,48 @@ Currently OTA  does not update your app while the app is open or minimized.
...
+### 3.1 Safely Pushing OTA Updates Without Redeploying {#safeToUpdate}
 
-It is good practice to *always* redeploy your Mendix App before pushing a new over the air update. However, releasing an OTA update without redeploying your Mendix App to Mendix Cloud in these cases is usually safe:
```
Comment (Adam-Dupaski) [inline content/howto/mobile/how-to-ota.md:40]: "App"
Using acronym

File: content/howto/mobile/how-to-ota.md [hunk 8]
```diff
@@ -23,43 +23,48 @@ Currently OTA  does not update your app while the app is open or minimized.
...
 
 * Style changes
-* Changes to static images, text, or other static assets
+* Static images, text, or other static asset changes
 * Layout changes
 * Nanoflow changes
 * JavaScript action changes
 * Widgets shipped with Mendix added or removed
 * A new custom Javascript-only widget or module was added
-* Non-destructive model changes (for more information, see the [Offline First Reference Guide](/refguide/offline-first))
```
Comment (Adam-Dupaski) [inline content/howto/mobile/how-to-ota.md:49]: Title of doc for cross-reference text

File: content/howto/mobile/how-to-ota.md [hunk 9]
```diff
@@ -23,43 +23,48 @@ Currently OTA  does not update your app while the app is open or minimized.
...
+* Non-destructive model changes (for more information, see [Offline-First](/refguide/offline-first))
+
+{{% todo %}}[Maybe unclear what 'Widgets shipped with Mendix added or removed" entails]{{% /todo %}}
 
 ### 3.2 When a Full Release Is Required
 
-If you have made any changes directly to your iOS or Android project, you will have to fully redeploy you app to the app stores for the changes to take effect. OTA updates do not suffice, and a full release is required, in the following cases:
```
Comment (Adam-Dupaski) [inline content/howto/mobile/how-to-ota.md:53]: Typo
Unnecessary commas

File: content/howto/mobile/how-to-ota.md [hunk 10]
```diff
@@ -23,43 +23,48 @@ Currently OTA  does not update your app while the app is open or minimized.
...
+If you have made any changes directly to your iOS or Android project, you will have to fully redeploy your app to the app stores for the changes to take effect. OTA updates do not suffice and a full release is required in the following cases:
 
 * The initial release of your app
-* A Mendix Studio Pro version upgrade that requires a new Native Template version
```
Comment (Adam-Dupaski) [inline content/howto/mobile/how-to-ota.md:56]: Cross-reference useful

File: content/howto/mobile/how-to-ota.md [hunk 11]
```diff
@@ -76,11 +83,11 @@ Next you must build new binaries with this capability toggled on, and then relea
 	{{% image_container width="350" %}}![OTA capability enabled](attachments/nbui/advanced-capabilities-ota.png){{% /image_container %}}
 
 1. Click **Save**.
-1.  Now build an application for distribution and make a note of the version number used:
```
Comment (Adam-Dupaski) [inline content/howto/mobile/how-to-ota.md:79]: "Now" and "then" etc.

File: content/howto/mobile/how-to-ota.md [hunk 12]
```diff
@@ -76,11 +83,11 @@ Next you must build new binaries with this capability toggled on, and then relea
 	{{% image_container width="350" %}}![OTA capability enabled](attachments/nbui/advanced-capabilities-ota.png){{% /image_container %}}
 
 1. Click **Save**.
-1.  Now build an application for distribution and make a note of the version number used: 
+1.  Build an application for distribution and make a note of the version number used: 
 
 	{{% image_container width="350" %}}![Build release app page](attachments/nbui/build-release-app-for-ota.png){{% /image_container %}}
 
-1. To make the OTA update functionality available to your users, please release the new binaries via the appropriate app stores. If you are testing the functionality you can now install the apps on your test devices.
```
Comment (Adam-Dupaski) [inline content/howto/mobile/how-to-ota.md:83]: "Please"

File: content/howto/mobile/how-to-ota.md [hunk 13]
```diff
@@ -127,44 +139,48 @@ To release a new version OTA, follow these steps:
...
 
 Before issuing OTA updates or releasing new versions, please read and understand the [Offline First](/refguide/offline-first) reference guide. It is important to understand the implications of offline first.
 
-Mendix Native Apps are offline first. This means you should be cautious when changing the following elements, and should avoid changing them if possible:
+{{% todo %}}[Third cross-reference of that doc and it's also a prereq - necessary here?]{{% /todo %}}
+
+Mendix native apps are offline first. This means you should be cautious when changing the following elements, and should avoid changing them if possible:
 
 * The navigation profile
-* An offline first entity, for example entity name changes, new entity relationships, and more
```
Comment (Adam-Dupaski) [inline content/howto/mobile/how-to-ota.md:137]: Fuller explanation needed for examples of changing an offline-first entity

File: content/howto/mobile/how-to-ota.md [hunk 14]
```diff
@@ -127,44 +139,48 @@ To release a new version OTA, follow these steps:
...
 
-This issue is independent from OTA updates and specific to offline apps. Your offline app runs a snapshot of your runtime's model locally. So as a Mendix Developer, you have to think twice before doing major chages that might make the app's state unrecoverable. In the example above the entity model was changed, and when the app attempted to synchronize it failed. This can create unrecoverable situations that will require a re-installation of the app, and can lead to data loss for unsynced data.
+This issue is independent from OTA updates and specific to offline apps. Your offline app runs a snapshot of your runtime's model locally. So as a Mendix developer, you have to think twice before doing major chages that might make the app's state unrecoverable. In the example above the entity model was changed, and when the app attempted to synchronize it failed. This can create unrecoverable situations that will require a re-installation of the app, and can lead to data loss for unsynced data.
 
-## 7 Transfer a CLI OTA-Compatible App to the Mendix Native Mobile Builder {#from-cli-to-ui}
```
Comment (Adam-Dupaski) [inline content/howto/mobile/how-to-ota.md:149]: Consistent gerunds/participles for section headings

File: content/howto/mobile/how-to-ota.md [hunk 15]
```diff
@@ -127,44 +139,48 @@ To release a new version OTA, follow these steps:
...
 
-The transition from the CLI to Mendix Native Mobile Builder for OTA supporting apps requires a few manual steps. These steps ensure you do not have to release your apps to the app stores again. 
+The transition from the CLI to Mendix Native Mobile Builder for OTA-supporting apps requires a few manual steps. These steps ensure you do not have to release your apps to the app stores again. 
 
-### 7.1 Gather the Required Information 
+### 7.1 Gathering the Required Information 
 
 1. Navigate to [App Center](https://appcenter.ms).
 1. While logged in, find the Android and iOS apps used for building your app.
-1. Check the URLs and note down the application ID as seen in the URLs. For example, in `https://appcenter.ms/users/user.name/apps/App-Android/distribute/code-push`, **App-Android** is the Android app's ID.
```
Comment (Adam-Dupaski) [inline content/howto/mobile/how-to-ota.md:157]: Consistent code formatting to refer to example

File: content/howto/mobile/how-to-ota.md [hunk 16]
```diff
@@ -127,44 +139,48 @@ To release a new version OTA, follow these steps:
...
+### 7.1 Gathering the Required Information 
 
 1. Navigate to [App Center](https://appcenter.ms).
 1. While logged in, find the Android and iOS apps used for building your app.
-1. Check the URLs and note down the application ID as seen in the URLs. For example, in `https://appcenter.ms/users/user.name/apps/App-Android/distribute/code-push`, **App-Android** is the Android app's ID. 
-1. If your app is built under an organization the URL might look like this: `https://appcenter.ms/orgs/org-name/apps/App-Android/distribute/code-push`. In this case, note the **org-name** as seen in the URL.
+1. Check the URLs and note down the application ID as seen in the URLs. For example, in `https://appcenter.ms/users/user.name/apps/App-Android/distribute/code-push`, the Android app's ID is `App-Android`.
+1. If your app is built under an organization the URL might look like this: `https://appcenter.ms/orgs/org-name/apps/App-Android/distribute/code-push`. In this case, note the `org-name` as seen in the URL.
 
-### 7.2 Move Your App to the Mendix Native Mobile Builder
```
Comment (Adam-Dupaski) [inline content/howto/mobile/how-to-ota.md:160]: More minimal heading

File: content/howto/mobile/how-to-ota.md [hunk 17]
```diff
@@ -127,44 +139,48 @@ To release a new version OTA, follow these steps:
...
+### 7.2 Moving Your App to Native Mobile Builder
 
 1. Launch the Mendix Native Mobile Builder for your project.
-1. If you have not yet completed the setup wizard, please complete it now.
```
Comment (Adam-Dupaski) [inline content/howto/mobile/how-to-ota.md:163]: "please"

File: content/howto/mobile/how-to-ota.md [hunk 18]
```diff
@@ -127,44 +139,48 @@ To release a new version OTA, follow these steps:
...
+1. If you have not yet completed the setup wizard, complete it now.
 1. Quit the tool completely. 
-1. Navigate to your app's directory and find the **nativemobile** folder (for example **C:\Users\user\Documents\Mendix\App\nativemobile**).
+1. Navigate to your app's directory and find the **nativemobile** folder (for example, **C:\Users\user\Documents\Mendix\App\nativemobile**).
 1. Enable **Hidden items** in Explorer to be able to see the *.config* file if it is not visible. 
-1.  Open the *.config* file using Notepad and look for a key named **App Center**. If it is there it might contain some app names already like this example:
```
Comment (Adam-Dupaski) [inline content/howto/mobile/how-to-ota.md:167]: Need to specify editor?

File: content/howto/mobile/how-to-ota.md [hunk 19]
```diff
@@ -183,17 +199,17 @@ The transition from the CLI to Mendix Native Mobile Builder for OTA supporting a
...
     ```
     
-    If the file does not exist add it manually. Either way, **make sure to change the names to reflect the IDs of your App Center apps**. Then make sure the newly changed file is still a valid JSON.
+    If the file does not exist, add it manually. Either way, **make sure to change the names to reflect the IDs of your App Center apps**. Then make sure the newly changed file is still a valid JSON.
     
 1. Restart the Mendix Native Mobile Builder for your project. If the tool does not start, verify once more that the *.config* file is a valid JSON.
 
 Try to push an OTA update for an unreleased version of your app, for example v0.1.0. If the OTA update shows up on your App Center app's CodePush administration page, congratulations! You successfully transferred your project over to the Mendix Native Mobile Builder.
 
-If the OTA release button remains disabled. Verify that the app names added in the configuration are correct and make sure to add the organization property if your apps are under an organization. After you corrected the mistakes, restart the Mendix Native Mobile Builder and try again.
```
Comment (Adam-Dupaski) [inline content/howto/mobile/how-to-ota.md:192]: Typo

File: content/howto/mobile/how-to-ota.md [hunk 20]
```diff
@@ -183,17 +199,17 @@ The transition from the CLI to Mendix Native Mobile Builder for OTA supporting a
...
+If the OTA release button remains disabled, verify that the app names added in the configuration are correct and make sure to add the organization property if your apps are under an organization. After you correct the mistakes, restart the Mendix Native Mobile Builder and try again.
 
 ## 8 Read More
 
 * [How to Deploy Your First Mendix Native Mobile App](/howto/mobile/deploying-native-app)
-* [Offline First Reference Guide](/refguide/offline-first)
-* [Codepush Introduction](https://docs.microsoft.com/en-us/appcenter/distribution/codepush/)
```
Comment (Adam-Dupaski) [inline content/howto/mobile/how-to-ota.md:198]: Typo

File: content/howto/mobile/how-to-ota.md (hunk 4 above)
Comment (Adam-Dupaski) [reply content/howto/mobile/how-to-ota.md:14]: Try to use verb to be more precise: "functions as an app"

File: content/howto/mobile/how-to-ota.md [hunk 21]
```diff
@@ -23,48 +23,55 @@ Currently OTA  does not update your app while the app is open or minimized.
...
 * The initial release of your app
-* A Mendix Studio Pro version upgrade that requires a new Native Template version
+* A Studio Pro version upgrade that requires a new [Native Template](/refguide/native-template) version
 * You fundamentally changed your app's functionality (this is an Apple App Store limitation, and will require a re-release and re-review of your app by Apple — your app might be removed if you do not comply)
 * A new native module has been added
 * The app has been renamed
 * The app's launcher icons have been changed
 * The splash screen has been changed
 
+{{% todo %}}[Example of native module that could be added?]{{% /todo %}}
```
Comment (Adam-Dupaski) [inline content/howto/mobile/how-to-ota.md:66]: Check if "native modules" used elsewhere in docs, add an example if possible

File: content/howto/mobile/how-to-ota.md [hunk 22]
```diff
@@ -127,44 +139,48 @@ To release a new version OTA, follow these steps:
 
 	{{% image_container width="300" %}}![Update success prompt](attachments/how-to-ota/phone-success-prompt.png){{% /image_container %}}
 
-## 6 Preserving your Model's Integrity
+## 6 Preserving Your Model's Integrity
 
 Before issuing OTA updates or releasing new versions, please read and understand the [Offline First](/refguide/offline-first) reference guide. It is important to understand the implications of offline first.
```
Comment (ConnorLand) [inline content/howto/mobile/how-to-ota.md:144]: Can delete this paragraph, as Offline-First is referenced elsewhere.

File: content/howto/mobile/how-to-ota.md [hunk 23]
```diff
@@ -127,44 +139,48 @@ To release a new version OTA, follow these steps:
...
+* An offline-first entity (for example, with entity name changes or new entity relationships)
 
-Generally, you should avoid doing destructive changes to offline-synced entities. In the rare cases this is unavoidable, releasing a new app version or doing over the air updates might put your app's users in an unrecoverable state.
+Generally, you should avoid making destructive changes to offline-synced entities. In the rare cases this is unavoidable, releasing a new app version or doing OTA updates might put your end-users in an unrecoverable state.
 
 ### 6.1 Offline Apps and Data Loss
 
 Data loss can occur when OTA updates or new releases coincide with apps being offline. For example, imagine your Mendix developers were hard at work optimizing the data store entity structure by consolidating entities to speed up sync operations. They release that morning. They push a new runtime by clicking the **Publish** button in Studio Pro, and then run the Native Mobile Builder to push a new update to the apps. All seems to work fine.
 
+{{% todo %}}[Unclear how "your Mendix developers" would differ from reader of doc in this scenario - is there a way to simplify this story?]{{% /todo %}}
```
Comment (Adam-Dupaski) [inline content/howto/mobile/how-to-ota.md:159]: Probably better to use "you"

**Review by ConnorLand — APPROVED** [review]

**Pull request merged**